addons/spreadsheet/static/src/@types/getters.d.ts

```diff
@@ -0,0 +1,72 @@
+import { CorePlugin, Model } from "@odoo/o-spreadsheet";
+import { ChartOdooMenuPlugin, OdooChartCorePlugin, OdooChartUIPlugin } from "@spreadsheet/chart";
+import { CurrencyPlugin } from "@spreadsheet/currency/plugins/currency";
+import {AccountingPlugin} from "addons/spreadsheet_account/static/src/plugins/accounting_plugin";
+import { GlobalFiltersCorePlugin, GlobalFiltersUIPlugin } from "@spreadsheet/global_filters";
+import { ListCorePlugin, ListUIPlugin } from "@spreadsheet/list";
+import { IrMenuPlugin } from "@spreadsheet/ir_ui_menu/ir_ui_menu_plugin";
+import { PivotCorePlugin, PivotUIPlugin } from "@spreadsheet/pivot";
+
+type Getters = Model["getters"]
+type CoreGetters = CorePlugin["getters"]
+
+/**
+ * Union of all getter names of a plugin.
+ *
+ * e.g. With the following plugin
+ * @example
+ * class MyPlugin {
+ *   static getters = [
+ *     "getCell",
+ *     "getCellValue",
+ *   ] as const;
+ *   getCell() { ... }
+ *   getCellValue() { ... }
+ * }
+ * type Names = GetterNames<typeof MyPlugin>
+ * // is equivalent to "getCell" | "getCellValue"
+ */
+type GetterNames<Plugin extends { getters: readonly string[] }> = Plugin["getters"][number];
+
+/**
+ * Extract getter methods from a plugin, based on its `getters` static array.
+ * @example
+ * class MyPlugin {
+ *   static getters = [
+ *     "getCell",
+ *     "getCellValue",
+ *   ] as const;
+ *   getCell() { ... }
+ *   getCellValue() { ... }
+ * }
+ * type MyPluginGetters = PluginGetters<typeof MyPlugin>;
+ * // MyPluginGetters is equivalent to:
+ * // {
+ * //   getCell: () => ...,
+ * //   getCellValue: () => ...,
+ * // }
+ */
+type PluginGetters<Plugin extends { new(...args: unknown[]): any; getters: readonly string[] }> = Pick<InstanceType<Plugin>, GetterNames<Plugin>>;
+
+declare module "@spreadsheet" {
+  /**
+   * Add getters from custom plugins defined in odoo
+   */
+
+  interface OdooCoreGetters extends CoreGetters { }
```

Comment:
Will that work if we try to extend it from a different odoo module ?

Reply:
Nope, we have to declare all of them there. It's the reason why we declare the ones of PivotUiPlugin for example.

But I think "le jeu en vaut la chandelle" :D